### ISSUE TITLE
Scss @include or @mixin with multiline parameters brokes afterwards messages

### ISSUE THREAD

**Issue by Tenkoru** [stylelint/stylelint#4699]
> Clearly describe the bug

If I use @include or @mixin with parameters at multiline it's breaks messages and highlighted errors, that goes afterwards.

> Which rule, if any, is the bug related to?

Not depends on rules

> What code is needed to reproduce the bug?

e.g.

```scss
.instance-button {
  @include button (
    $padding: 0 0 0 16px,
    $border: none,
    $background-color: transparent,
    $text-transform: capitalize,
    $fill: $black-l-5);
  .d-Icon {
    margin-right: 16px;
  }
}
```

> What stylelint configuration is needed to reproduce the bug?

e.g.

Not depends on rules

> Which version of stylelint are you using?

e.g. `13.3.1`

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

e.g. "Webstorm, also tryed in Intelij Idea"

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

e.g. "Yes, it's related to SCSS nested properties."

> What did you expect to happen?

e.g. "Messages and highlights at right places"

> What actually happened (e.g. what warnings or errors did you get)?

e.g. "The following warnings were flagged:"

```shell
No warnont
```

Two screenshots to show the difference:
![image](https://user-images.githubusercontent.com/13865243/78942043-85c86100-7ac1-11ea-85bf-86a243dbb5b4.png)
![image](https://user-images.githubusercontent.com/13865243/78942073-97aa0400-7ac1-11ea-99b7-802c3cf23de6.png)

**Comment by jeddy3** [comment]
@Tenkoru Thanks for the report and for using the template.

I'm unable to reproduce this using [the demo](https://stylelint.io/demo#N4Igxg9gJgpiBcIB0BLAdgZwC4EM1hgFoAjAVyywjQAJgAdG6gAXTABtTZqyKrqAKBtWHUAJAAccUKOgDm8agAYlKgIwA2cQA8ANEJGjiEAE6xjCtFRh7GwwzjABrWcYik0UQpDYmFWY3gYksYwaFg2ImJYMFpYhP6BAGYmALYKYDjiKLhsKABe1vp2iShsbAqGbA6OhGyEAKwAlADc+kieAJKQNPS21Ck4xrLohMYosgAWWAoa2q2MAL4MS2gMSMRShNgAnmwwtAsgOiAY22E4WggnYBgYR+BUJbJXvcJ0IMakexjvCq8i72IPichEshBgKXEWG2v2o-lIMH0KxWIAWQA), the CLI nor the VS Code plugin. They correctly report that the violation is on line 13.

The problem may lie with the Webstorm and Intelij Idea plugin. I recommend:

- checking you're using the latest version of stylelint (currently `13.3.1`)
- raising a bug report in their issue tracker

I'm going to close this issue for now as it doesn't appear to be a problem with stylelint itself. Feel free to reopen with a reproducible example, though.

**Comment by Tenkoru** [comment]
Thanks for answer. I created a issue at jetbrains forum.
https://youtrack.jetbrains.com/issue/WEB-44856
What i should do next?

**Comment by jeddy3** [comment]
> Elena Pogorelova commented 2 hours ago
> didn't try this in VSCode, but I can easily reproduce the issue by running stylelint in terminal. So it's clearly not an issue with Webstorm

I'm unable to reproduce it locally. 

```scss
.instance-button {
  @include button (
    $padding: 0 0 0 16px,
    $border: none,
    $background-color: transparent,
    $text-transform: capitalize,
    $fill: $black-l-5);
  .d-Icon {
    margin-right: 16px;
  }
}

.bad-style {}
```

```json
{
  "rules": {
    "block-no-empty": true
  }
}
```

Gives me:

```console
jeddy3@UNKNOWN stylelint-test % npx stylelint "sources/scss/fix.scss" --config configs/fix.json

sources/scss/fix.scss
 13:12  ✖  Unexpected empty block   block-no-empty
```

However, Elena Pogorelova was able to reproduce it. Therefore something is definitely amiss. I'll reopen the issue as needing investigation.

@Tenkoru Can you:

- try reproducing the issue locally using the stylelint CLI?
- tell us what OS are you using, e.g. MacOS, Windows 10 etc?

A difference in OS is usually the culprit with these mysterious bugs.

**Comment by Tenkoru** [comment]
I reproduced it localy with CLI.
My OS is Windows 10
https://prnt.sc/rydrf6

**Comment by jeddy3** [comment]
Thanks for reproducing it locally. This is a bug in stylelint, but specific to Windows 10. 

The problem lies in either the [`block-no-empty` rule](https://github.com/stylelint/stylelint/blob/master/lib/rules/block-no-empty/index.js) (perhaps where it calculates the index) or upstream [in the parser](https://github.com/postcss/postcss-scss) stylelint uses [to support SCSS](https://stylelint.io/about/syntaxes). 

I've labelled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide.

**Comment by Tenkoru** [comment]
Unfortunately, problem exists with any rule, than goes after multiline parameters.

**Comment by jeddy3** [comment]
That hints at it being a parser issue, although it doesn't guarantee it. Resolving the bug requires a small investigation to first pinpoint where the issue lies. Please consider digging deeper if you have time, and [contributing to the parser](https://stylelint.io/developer-guide/syntaxes#working-on-syntaxes) if the problem lies there.
